test/UnitTests/MSTest.Analyzers.UnitTests/AssertionArgsShouldBePassedInCorrectOrderAnalyzerTests.cs

```diff
@@ -889,4 +889,112 @@ public void Compliant()
 
         await VerifyCS.VerifyCodeFixAsync(code, code);
     }
+
+    [TestMethod]
+    public async Task TypeOfExpressions()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            using System;
+            using System.IO;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void NonCompliant()
+                {
+                    Type actualType = typeof(string);
+                    Exception exception = new IOException();
+
+                    // These should be flagged - typeof() expressions should be treated as constants
+                    [|Assert.AreEqual(actualType, typeof(string))|];
+                    [|Assert.AreEqual(exception.GetType(), typeof(IOException))|];
+                    [|Assert.AreNotEqual(actualType, typeof(int))|];
+                    [|Assert.AreSame(actualType, typeof(string))|];
+                    [|Assert.AreNotSame(actualType, typeof(int))|];
+                }
+
+                [TestMethod]
+                public void Compliant()
+                {
+                    Type actualType = typeof(string);
+                    Exception exception = new IOException();
+
+                    // These are correct - typeof() expressions as expected values
+                    Assert.AreEqual(typeof(string), actualType);
+                    Assert.AreEqual(typeof(IOException), exception.GetType());
+                    Assert.AreNotEqual(typeof(int), actualType);
+                    Assert.AreSame(typeof(string), actualType);
+                    Assert.AreNotSame(typeof(int), actualType);
+                }
+            }
+            """;
+
+        string fixedCode = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            using System;
+            using System.IO;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void NonCompliant()
+                {
+                    Type actualType = typeof(string);
+                    Exception exception = new IOException();
+
+                    // These should be flagged - typeof() expressions should be treated as constants
+                    Assert.AreEqual(typeof(string), actualType);
+                    Assert.AreEqual(typeof(IOException), exception.GetType());
+                    Assert.AreNotEqual(typeof(int), actualType);
+                    Assert.AreSame(typeof(string), actualType);
+                    Assert.AreNotSame(typeof(int), actualType);
+                }
+
+                [TestMethod]
+                public void Compliant()
+                {
+                    Type actualType = typeof(string);
+                    Exception exception = new IOException();
+
+                    // These are correct - typeof() expressions as expected values
```

Comment:
These comments in the fixed code are IMHO more confusing than helpful, this is 3-rd time that I am spending time to figure out what is wrong with the code, only to later realize that I am looking at the fixed code and the comment is no longer accurate.

Reply:
They make sense to me but I don't mind if we keep or remove them.